src/editor/components/modals/revision-list.jsx

```diff
@@ -75,11 +89,26 @@ class RevisionListUnconnected extends React.Component {
 
 export function mapStateToProps(state) {
   const notebookHistory = state.notebookHistory || {};
-  const { revisionList, selectedRevisionId } = notebookHistory;
+  const {
+    revisionContentFetchStatus,
+    revisionContent,
+    revisionList,
+    selectedRevisionId
+  } = notebookHistory;
+
+  let hideLocalChanges = true;
+
+  if (revisionContentFetchStatus === "IDLE") {
+    const previousRevisionContent =
+      revisionList.length > 0 ? revisionContent[revisionList[0].id] : "";
+
+    hideLocalChanges = state.iomd === previousRevisionContent;
+  }
```

Comment:
I think we could rewrite this as something like this:

```js
const haveLocalChanges = revisionList && revisionList.length && state.iomd !== previousRevisionContent;
```

At which point you could just include it below as a parameter to the object you're returning: 

```js
return {
  revisionList,
  ...
  haveLocalChanges: revisionList && revisionList.length && state.iomd !== previousRevisionContent
}
```